lib/al/Library/Area/AreaShapeSphere.cpp

```diff
@@ -0,0 +1,45 @@
+#include "al/Library/Area/AreaShapeSphere.h"
+
+namespace al {
+AreaShapeSphere::AreaShapeSphere() {}
+
+bool AreaShapeSphere::isInVolume(const sead::Vector3f& trans) const {
+    sead::Vector3f baseTrans;
+    calcTrans(&baseTrans);
+    sead::Vector3f offsetTrans = trans - baseTrans;
+    f32 radius = mScale.x * 500.0f;
+
+    return offsetTrans.squaredLength() <= radius * radius;
```

Comment:
```suggestion
    return offsetTrans.squaredLength() <= sead::Mathf::square(getScale().x * 500);
```

Reply:
this causes a mismatch, seen here: https://decomp.me/scratch/HoRXn

Reply:
Alright, then you should to with the two-lined variant, but still use `square` instead:
https://decomp.me/scratch/0HdGu